src/socket.ts

```diff
@@ -0,0 +1,20 @@
+import { Socket } from 'net'
+import { AbqConfiguration } from './configuration'
+
+export async function connect(abqConfig: AbqConfiguration): Promise<Socket> {
+  if (!abqConfig.enabled) {
+    throw new Error('abq must be enabled to connect')
+  }
+
+  const socket = new Socket()
+
+  return await new Promise(resolve => {
```

Comment:
💭 wonder if this should also handle the spawnedMessage call

Reply:
This package is _very_ low-level right now. I think I agree though... we should review once we have two consumers of it to hone in on what the right level of abstraction is.